cpp/oneapi/dal/algo/kmeans/backend/gpu/train_kernel_lloyd_dense_dpc.cpp

```diff
@@ -96,65 +95,153 @@ struct train_kernel_gpu<Float, method::lloyd_dense, task::clustering> {
     train_result<task::clustering> operator()(const dal::backend::context_gpu& ctx,
                                               const descriptor_t& params,
                                               const train_input<task::clustering>& input) const {
-        const auto data = input.get_data();
+        auto& queue = ctx.get_queue();
 
+        const auto data = input.get_data();
         const int64_t row_count = data.get_row_count();
         const int64_t column_count = data.get_column_count();
+        auto data_ptr =
+            row_accessor<const Float>(data).pull(queue, { 0, -1 }, sycl::usm::alloc::device);
+        auto arr_data = pr::ndarray<Float, 2>::wrap(data_ptr, { row_count, column_count });
 
         const int64_t cluster_count = params.get_cluster_count();
         const int64_t max_iteration_count = params.get_max_iteration_count();
         const double accuracy_threshold = params.get_accuracy_threshold();
 
-        auto daal_initial_centroids = get_initial_centroids<Float>(ctx, params, input);
-
-        auto& queue = ctx.get_queue();
-        interop::execution_context_guard guard(queue);
-
-        daal_kmeans::Parameter par(dal::detail::integral_cast<std::size_t>(cluster_count),
-                                   dal::detail::integral_cast<std::size_t>(max_iteration_count));
-        par.accuracyThreshold = accuracy_threshold;
-
-        const auto daal_data = interop::convert_to_daal_table(queue, data);
+        auto initial_centroids = get_initial_centroids<Float>(ctx, params, input);
+        daal::data_management::BlockDescriptor<Float> block;
+        initial_centroids->getBlockOfRows(0, cluster_count, daal::data_management::readOnly, block);
+        Float* initial_centroids_ptr = block.getBlockPtr();
+        auto arr_initial =
+            pr::ndarray<Float, 2>::wrap(initial_centroids_ptr, { cluster_count, column_count });
```

Comment:
Update `get_initial_centroids` function to return ready-to-use ndarray with the initial centroids, but not `NumericTable`. Please double check that `arr_initial` is allocated on device.

Reply:
Done